app/src/main/java/pm/gnosis/heimdall/ui/walletconnect/intro/WalletConnectIntroActivity.kt

```diff
@@ -0,0 +1,109 @@
+package pm.gnosis.heimdall.ui.walletconnect.intro
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.viewpager.widget.PagerAdapter
+import androidx.viewpager.widget.ViewPager
+import com.jakewharton.rxbinding2.view.clicks
+import io.reactivex.Single
+import io.reactivex.rxkotlin.plusAssign
+import io.reactivex.rxkotlin.subscribeBy
+import kotlinx.android.synthetic.main.layout_wallet_connect_intro.*
+import pm.gnosis.heimdall.R
+import pm.gnosis.heimdall.di.components.ViewComponent
+import pm.gnosis.heimdall.reporting.ScreenId
+import pm.gnosis.heimdall.ui.base.BaseActivity
+import pm.gnosis.heimdall.ui.base.ViewModelActivity
+import pm.gnosis.heimdall.ui.walletconnect.sessions.WalletConnectSessionsActivity
+import pm.gnosis.model.Solidity
+import pm.gnosis.utils.asEthereumAddress
+import pm.gnosis.utils.asEthereumAddressString
+
+
+class WalletConnectIntroActivity : ViewModelActivity<WalletConnectIntroContract>() {
+
+    private val layouts =
```

Comment:
better to use enum for intro page, as enum values can potentially hold additional useful data if needed. Also it will be more readable

Reply:
wouldn't that be unnecessary overhead here (since we don't have any other information right now?)